File: Leetcode - Java/167_array_twoSum2.java
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] res = {-1,-1};
        int low = 0, high = numbers.length-1;
        
        while(low < high) {
            int sum = numbers[low] + numbers[high];
            if(sum == target) {
                res[0] = low+1;
                res[1] = high+1;
                return res;
            }
            else if(sum < target) ++low;
            else --high;
        }
        return res; 
    }
}

File: Leetcode - C++/167_array_twoSum2.cpp
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
// Time - O(N), Space - O(N)

#include <iostream>
#include <vector>

using namespace std; 

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        unordered_map<int, int> map;
        vector<int> res;
        for(int i = 0; i < numbers.size(); i++) {
            int num = target - numbers[i];
            if(map.find(num) != map.end()) {
                res.push_back(map[num]);
                res.push_back(i+1);
                return res;
            }
            map[numbers[i]] = i+1;
        }
        return vector<int>();
    }
};

int main() {
    Solution *solution = new Solution();
    vector<int> numbers = {2,7,11,15};
    int target = 9;
    vector<int> res = solution->twoSum(numbers, target);
    for(int n : res)
        cout << n << " ";
    cout << endl;
}